### PR TITLE
E2E tests creating cluster with latest release & upgrading with main

### PR DESCRIPTION
**Pull request by taneyland**
*Resolves* https://github.com/aws/eks-anywhere/issues/324

*Description of changes:*
* Run upgrade flow e2e tests by creating the cluster with the latest minor release and upgrading with current code
* Latest release is dynamic, it uses the `T_BRANCH_NAME` env var to determine the branch where the tests are being run:
  * If in `main`, this is the latest available eksa release (eg. `v0.6.1`)
  * If in a release branch, this is the latest release from the previous minor version (eg. for `release-0.6`, `v0.5.0`)
* Removes paths from the cluster yaml config to avoid validation errors from old CLIs that don't know how to read new api fields. This is a bit hacky and it involves maintaining a list of incompatible fields in code. Ideally, we should refactor the api fillers package to not use the latest api structs. That's a considerable effort, so I went with this temporal solution.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/upgrade_test.go [hunk 1]
```diff
@@ -32,6 +32,15 @@ func runSimpleUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 	test.DeleteCluster()
 }
 
+func runUpgradeFlowDiffCliVersionFromMain(test *framework.ClusterE2ETest, updateVersion v1alpha1.KubernetesVersion, clusterOpts ...framework.ClusterE2ETestOpt) {
```
Comment (g-gaston) [inline test/e2e/upgrade_test.go:35]: I'm ok with this for now, but shouldn't we make this dynamic so it works in release branches as well?
Like if the current branch is main, it uses `ExecuteWithLatestMinorReleaseFromMain`
But if it's a release branch, it uses something different

File: test/framework/releaseVersions.go [hunk 2]
```diff
@@ -153,3 +139,21 @@ func prodReleases() (release *releasev1alpha1.Release, err error) {
 	}
 	return releases, nil
 }
+
+func getLatestPrevMinorRelease(releases *releasev1alpha1.Release, releaseBranchVersion *semver.Version, targetRelease *releasev1alpha1.EksARelease) (*releasev1alpha1.EksARelease, error) {
```
Comment (g-gaston) [inline test/framework/releaseVersions.go:143]: why pass `targetRelease` as an argument and then return it as well?
I would vote to remover the assignment in this method and just return it

File: test/e2e/upgrade_test.go (hunk 1 above)
Comment (taneyland) [reply test/e2e/upgrade_test.go:35]: Yeah, good point. I was originally going to have a separate function for the release branch tests called `runUpgradeFlowDiffCliVersionFromReleaseBranch`. And in that one, we would use the method that we already have in place called `ExecuteWithLatestMinorReleaseFromVersion`.

But making it dynamic seems to make more sense so when I add the tests for release branches in a separate PR, I will look into making this dynamic

**Comment by danbudris** [comment]
https://github.com/aws/eks-anywhere/issues/324

**Review by danbudris — APPROVED** [review]

**Pull request merged**